oauthproxy.go

```diff
@@ -269,6 +271,11 @@ func (p *OAuthProxy) setupServer(opts *options.Options) error {
 		TLS:               opts.Server.TLS,
 	}
 
+	// Option: AllowQuerySemicolons
+	if opts.AllowQuerySemicolons {
+		serverOpts.Handler = http.AllowQuerySemicolons(p)
```

Comment:
To make this a little more futureproof/obvious you are just adding a wrapper, perhaps you should pass in `serverOpts.Handler` instead of `p`?

Is this the right place to do this anyway? We have a number of HTTP handler chains being set up anyway, have you considered adding it into that instead?

Reply:
It can possibly be done there as well, depends on your preference - If you prefer it to be contained in a chain, let me know and I can re-submit the pull.

Reply:
Yeah I think lets change it to chain and then in theory we don't need to update it later if someone injects additional handlers into this chain

Reply:
@JoelSpeed I'm not sure @MickMake is going to get back to this as he was working on it for us at the time. While I can make the initial change suggestion (passing `serverOpts.Handler` instead of `p`), I'm not sure where the chain changes should be made. Can you point me to where the chain of HTTP Handlers are so I can try and update this PR in the desired style?